app/controllers/search_controller.rb

```diff
@@ -123,6 +123,21 @@ def find_material(type, id)
     return material
   end
 
+  def get_current_material_anonymous
+    material_type = params[:material_type]
+    material_ids = params[:material_id]
+    material_ids = material_ids.split(',')
+
+    if material_ids.length == 1 #Check if material to be assigned is a single activity or investigation
+      @material = [find_material(material_type, params[:material_id])]
+    else
+      @material = material_ids.collect{|a| ::Activity.find(a)}
```

Comment:
This looks broken. It the material_type is not Activity then the wrong material would be found.
Also we don't really use Activity anymore directly so I'd think this should probably be ExternalActivity.  But maybe it would better to remove the if and do a collect with find_materials like:`@material = material_ids.collect{|matId| find_material(material_type, matId)}`
Or maybe this route only really needs to handle a single material not a list?

Reply:
@scytacki: I think this would only ever handle a single material, but I'm not sure. It does seem to suggest that it's possible that multiple materials could be handled, but I don't know of anywhere on the portal where you can assign multiple materials at once. Is that a feature somewhere?

I essentially copied this from the `get_current_material_unassigned_clazzes` method (lines 141 - 170). The use of Activity came from there. Is that in need of being fixed, too? Maybe it hasn't ever caused problems because it's not currently possible to assign more than one material at a time?

Looking at this again, I'm wondering if it'd be better to merge those two methods (`get_current_material_anonymous` and `get_current_material_unassigned_clazzes`). Do you think that would be better if it's possible?